### ISSUE TITLE
Taxonomy vocabularies are randomly ordered when created and their order is not reflected on the admin bar.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#2657]
I noticed this while testing the latest PR for #2247 ...

![image](https://cloud.githubusercontent.com/assets/2423362/25549073/1a94a308-2cb3-11e7-8b0b-ff487f65f903.png)

The vocabularies were added in this order: abc -> 123 -> xyz

Order in the taxonomy list page: xyz -> Tags -> abc -> 123
Order on the admin bar menu: the reverse

**Comment by klonos** [comment]
...I can't think of a reason why we should be ordering these in a specific way or another (especially since that list can be manually reordered), but I imagine that the end user expects this to be consistent:

- If we choose to leave the last-added vocabulary last, then do so with every vocabulary added thereafter.
- Have the order in the admin bar be consistent with the order in the list.

**Comment by olafgrabienski** [comment]
I'd say let's list them alphabetical, like taxonomy terms (which can be reordered as well), or like content types. If there's no special reason to choose another way of sorting, alphabetical has the best usability.

**Comment by Dinornis** [comment]
👍 for alphabetical ordering.

**Comment by klonos** [comment]
Yeah, alphabetical makes sense where things cannot be ordered (like the content type list for example). Here though, if a user changes the ordering of one of the vocabularies, then trying to retain alphabetical order simply stops making sense after that. Let me explain by example...

Say we do list things alphabetically. By default, we have the "Tags" vocabulary. Lets say that a user adds a "123", an "abc" and an "XYZ" vocabularies. The resulting order would be:

123
abc
Tags
XYZ

Now, say the user reorders the vocabularies like so:

abc
XYZ
Tags
123

Say the next try to add a "456" vocabulary. Where do we add this when created?

Say they want to add a "Random" vocabulary. If the order has not been changed and we were still listing in alpha, we'd do this:

123
abc
Tags
Random
XYZ

...but with the vocabularies manually reordered, where does it belong?

That's why I think it makes better sense to always have the newly-created vocabularies added to the bottom of that list. That way, users will now where to look for them and they can drag them in the order they wish. It's the same pattern we use when adding new fields.

There is the other argument though: after a new vocabulary is created, the user is redirected to the add terms form. Once they are finished and they navigate back to the list of vocabularies, if there are a lot of vocabularies and we add the new one at the bottom, the user might then need to scroll in order to see it. This can be mitigated if we just add a "new" indicator to any vocabularies that were added to this list after the last time the user has been here. In fact that is a good idea to do in general with newly-added items on lists (but also a separate issue 😄 ).

No matter what we decide though, the order of the vocabularies on this list should be reflected on the admin bar.

**Comment by olafgrabienski** [comment]
I've no strong opinion about how to sort the vocabularies by default on the taxonomy page. That said, I'm still pro default alphabetical because the order is to be reflected on the Admin bar. If the user doesn't care about reordering the vocabularies -- and I guess that's often the case -- an alphabetical list is still much better scannable / readable, and with the Admin bar in question there are even two places where to consider the better readability.
